### PR TITLE
[SYCL][Matrix] Add matrix combinations 16x16x16 and 32x64x16 bfloat16 for PVC

### PR DESCRIPTION
**Pull request by dkhaldi**

### CODE REVIEW

File: sycl/source/detail/device_info.hpp [hunk 1]
```diff
@@ -763,6 +763,10 @@ struct get_device_info_impl<
            matrix_type::fp32, matrix_type::fp32},
           {8, 0, 0, 0, 16, 16, matrix_type::bf16, matrix_type::bf16,
            matrix_type::fp32, matrix_type::fp32},
+          {0, 0, 0, 16, 16, 16, matrix_type::bf16, matrix_type::bf16,
```
Comment (dm-vodopyanov) [inline sycl/source/detail/device_info.hpp:766]: Please also update https://github.com/intel/llvm/blob/sycl/sycl/test-e2e/Matrix/runtime_query_pvc.cpp

File: sycl/source/detail/device_info.hpp [hunk 2]
```diff
@@ -763,6 +763,10 @@ struct get_device_info_impl<
...
+           matrix_type::fp32, matrix_type::fp32},
```
Comment (dm-vodopyanov) [inline sycl/source/detail/device_info.hpp:767]: Please also update the spec

File: sycl/source/detail/device_info.hpp (hunk 1 above)
Comment (dkhaldi) [reply sycl/source/detail/device_info.hpp:766]: done

File: sycl/source/detail/device_info.hpp (hunk 2 above)
Comment (dkhaldi) [reply sycl/source/detail/device_info.hpp:767]: done

**Review by dm-vodopyanov — APPROVED** [review]

**Review by YuriPlyakhin — APPROVED** [review]
LGTM

**Review by gmlueck — APPROVED** [review]
spec changes LGTM.

**Comment by YuriPlyakhin** [comment]
@intel/llvm-gatekeepers, all green, please merge

**Pull request merged**